gtsam/basis/ParameterMatrix.h

```diff
@@ -79,29 +79,27 @@ class ParameterMatrix {
   }
 
   /**
-   * Get the matrix column specified by `index`.
-   * @param index: The column index to retrieve.
+   * Set the matrix row specified by `index`.
+   * @param index: The row index to set.
    */
-  Eigen::Matrix<double, M, 1> col(size_t index) const {
-    return matrix_.col(index);
+  auto row(size_t index) -> Eigen::Block<MatrixType, 1, -1, false> {
+    return matrix_.row(index);
   }
 
   /**
-   * Set the matrix row specified by `index`.
-   * @param index: The row index to set.
-   * @param row: The Eigen (row) vector used to set the values.
+   * Get the matrix column specified by `index`.
+   * @param index: The column index to retrieve.
    */
-  void setRow(size_t index, const Eigen::Matrix<double, 1, -1>& row) {
-    matrix_.row(index) = row;
+  Eigen::Matrix<double, M, 1> col(size_t index) const {
+    return matrix_.col(index);
   }
```

Comment:
Would this work? Isn't p2 now const?

Reply:
Ya, so what'll happen is `p2.row(r)` will return a temporary Matrix (copy) and that temporary matrix is modified then immediately discarded.

With stricter compile flags conceivably it could throw unused warnings or something, but this passed on my machine with default gtsam settings.

In any case, in retrospect after your comment about the purpose of ParameterMatrix, it's a lot more obvious these are just mimicking Eigen's behavior so I don't think this is needed anymore.